javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,456 @@
+        public int lastIndexOf(Object obj) {
+            @SuppressWarnings("unchecked") final T that = (T) obj;
+            return getDelegate().lastIndexOf(that);
+        }
+
+        @Override
+        public java.util.ListIterator<T> listIterator() {
+            return new ListIterator<>(this, 0);
+        }
+
+        @Override
+        public java.util.ListIterator<T> listIterator(int index) {
+            checkBounds(index);
+            return new ListIterator<>(this, index);
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public T remove(int index) {
+            return setDelegateAndGetPreviousElement(index, () -> (C) getDelegate().removeAt(index));
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public boolean remove(Object obj) {
+            @SuppressWarnings("unchecked") final T that = (T) obj;
+            return setDelegateAndCheckChanged(() -> (C) getDelegate().remove(that));
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public boolean removeAll(Collection<?> collection) {
+            Objects.requireNonNull(collection, "collection is null");
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) collection;
+            return setDelegateAndCheckChanged(() -> (C) getDelegate().removeAll(that));
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public boolean retainAll(Collection<?> collection) {
+            Objects.requireNonNull(collection, "collection is null");
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) collection;
+            return setDelegateAndCheckChanged(() -> (C) getDelegate().retainAll(that));
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public T set(int index, T element) {
+            return setDelegateAndGetPreviousElement(index, () -> (C) getDelegate().update(index, element));
+        }
+
+        @Override
+        public int size() {
+            return getDelegate().size();
+        }
+
+        @Override
+        public List<T> subList(int fromIndex, int toIndex) {
+            return new ListView<>(getDelegate().subSequence(fromIndex, toIndex), isMutable());
+        }
+
+        @Override
+        public Object[] toArray() {
+            return getDelegate().toJavaArray();
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <U> U[] toArray(U[] array) {
+            Objects.requireNonNull(array, "array is null");
+            final U[] target;
+            final int length = getDelegate().length();
+            if (array.length < length) {
+                final Class<? extends Object[]> newType = array.getClass();
+                target = (newType == Object[].class)
+                         ? (U[]) new Object[length]
+                         : (U[]) java.lang.reflect.Array.newInstance(newType.getComponentType(), length);
```

Comment:
Yes, will do. Have to burn it into my brain windings that relection isn't gwt compatible :)